src/sections/selectNode.tsx

```diff
@@ -20,51 +21,70 @@ function Section1() {
     (store: Store) => store.auth.nodes
   ).map((elem: any, index: number) => {
     return {
-      name: elem.apiEndpoint,
+      name: elem.localName ?  `${elem.localName} (${elem.apiEndpoint})` : elem.apiEndpoint,
+      localName: elem.localName,
       value: index,
       apiEndpoint: elem.apiEndpoint,
       apiToken: elem.apiToken,
     };
   });
-  const connecting = useAppSelector(
-    (store: Store) => store.auth.status.connecting
-  );
-  const connected = useAppSelector(
-    (store: Store) => store.auth.status.connected
-  );
+  const connecting = useAppSelector((store: Store) => store.auth.status.connecting);
+  const connected = useAppSelector((store: Store) => store.auth.status.connected);
+  const loginData = useAppSelector((store: Store) => store.auth.loginData);
+  
+  const [searchParams, setSearchParams] = useSearchParams();
+  const [localName, set_localName] = useState(loginData.localName ? loginData.localName : '');
+  const [apiEndpoint, set_apiEndpoint] = useState(loginData.apiEndpoint ? loginData.apiEndpoint : '');
+  const [apiToken, set_apiToken] = useState(loginData.apiToken ? loginData.apiToken : '');
+  const [saveApiToken, set_saveApiToken] = useState(loginData.apiToken ? true : false);
+  const [nodesSavedLocallyChosenIndex, set_nodesSavedLocallyChosenIndex] = useState('' as number | '');
+
+  useEffect(()=>{
+    // Update the Select based on loginData from the Store
+    if(!loginData.apiEndpoint) return;
+    const existingItem = nodesSavedLocally.findIndex((item: any) => (item.apiEndpoint === loginData.apiEndpoint));
```

Comment:
Do we have types for this? so we know what `item` has instead of any

Reply:
We can add them, please do it